src/directives/offense/poisonRoom.ts

```diff
@@ -1,169 +1,188 @@
+		if (Cartographer.roomType(this.pos.roomName) != ROOMTYPE_CONTROLLER) {
+			log.warning(`${this.print}: ${printRoomName(this.pos.roomName)} is not a controller room; ` +
+						`removing directive!`);
+			this.remove(true);
+		}
+		// failsafe - if already owned and controller level > 2, remove flag
+		if(this.room && this.room.controller && this.room.controller.my && this.room.controller.level > 2) {
+			this.remove(true);
+		}
+		// Remove if owned by other player
+		if (this.room && !!this.room.owner && this.room.owner != MY_USERNAME) {
+			log.notify(`Removing poisonRoom directive in ${this.pos.roomName}: room already owned by another player.`);
+			this.remove(true);
+		}
+		// Remove if not enough GCL
+		if(getAllColonies().length == Game.gcl.level) {
+			log.warning(`${this.print}: ${printRoomName(this.pos.roomName)} not enough GCL; ` +
+						`for poison room, removing directive!`);
+		}
+	}
+	spawnMoarOverlords() {
+		this.overlords.claim = new ClaimingOverlord(this);
+		this.overlords.roomPoisoner = new RoomPoisonerOverlord(this);
+	}
+	init() {
+		this.alert(`Poisoning Room ${this.pos.roomName}`);
+
+		if(!(this.room && this.room.controller)) {
+			return;
+		}
+
+		// suspend room immediately once claimed to avoid default room actions
+		if(this.room.controller.level == 1) {
+			this.suspendColony(this.room.name);
+		}
+
+		if (Game.time % RUN_TIMER == 0 || this.room.constructionSites.length == 0) {
+			// capture variables
+			this.walkableSourcePosisions	 = _.filter(_.flatten(_.map(this.room.sources,
+													s => s.pos.neighbors)),pos => pos.isWalkable(true));
+			this.walkableControllerPosisions =  _.filter(this.room.controller!.pos.neighbors,
+													pos => pos.isWalkable(true));
+			
+			// reverse reservation if needed
+			if(this.room.controller.reservation && this.room.controller.reservation.ticksToEnd > 500) {
+				DirectiveControllerAttack.createIfNotPresent(this.room.controller.pos,'room');
+			}
+
+			// send fighters if needed
+			if(this.room && this.room.playerHostiles.length > 0) {
+				DirectiveOutpostDefense.createIfNotPresent(this.room.controller.pos,'room');
+			}
+		}
+	}
+	private suspendColony(roomName: string): string {
+		if (Overmind.colonies[roomName]) {
+			const colonyMemory = Memory.colonies[roomName] as ColonyMemory | undefined;
+			if (colonyMemory) {
+				colonyMemory.suspend = true;
+				Overmind.shouldBuild = true;
+				return `Colony ${roomName} suspended.`;
+			} else {
+				return `No colony memory for ${roomName}!`;
+			}
+		} else {
+			return `Colony ${roomName} is not a valid colony!`;
+		}
+	}
+
+	private poisonActions() {
+		if(!(this.room && this.room.controller && this.room.controller.level == 2 && 
+			 this.walkableControllerPosisions && this.walkableControllerPosisions)) {
+			return;
+		}
+
+		// Poison Controller First
+		if(this.walkableControllerPosisions.length > 0) {
+			_.first(this.walkableControllerPosisions).createConstructionSite(STRUCTURE_WALL);
```

Comment:
What is the point of the first part? Don't you want to do it for all positions?

Reply:
I try to avoid placing multiple wall csites because they are not walkable, just in case if there unknown corner cases in room layout.
the priority is for controller. sources need to be last for obvious reasons